Src/library/net/Clib/syncpoll.c

```diff
@@ -107,15 +107,27 @@ EIF_INTEGER c_select_poll(EIF_INTEGER fd)
 
 EIF_INTEGER c_select_poll_with_timeout(EIF_INTEGER fd, 
 		                               EIF_BOOLEAN read_mode,
-									   EIF_INTEGER timeout)
-	/*x Get read/write status for socket fd within `timeout' seconds */
+									   EIF_NATURAL_64 timeout_nano_sec)
+	/*x Get read/write status for socket fd within `timeout_nano_sec` nanoseconds*/
 {
 	fd_set fdmask;
 	struct timeval tmout;
 	int res;
-
-	tmout.tv_sec = (unsigned long) timeout;
-	tmout.tv_usec = (long) 0;
+	EIF_NATURAL_64 sec;
+	sec = timeout_nano_sec / RTU64C(1000000000);
+	if (sec > LONG_MAX) {
+		/* Overflown, use max value instead */
+		tmout.tv_sec = (long) LONG_MAX;
+	} else {
+		tmout.tv_sec = (long) sec;
+	}
+	tmout.tv_usec = (long) (timeout_nano_sec % RTU64C(1000)); /* 1 microsecond = 1000 nanoseconds */
```

Comment:
Same issue here - we get nanoseconds that don't fit into a microsecond.

Reply:
using modulo % ... the result of `timeout_nano_sec % RTU64C(1000)` will always be between 0 and 1000.
so I guess it will fit in tv_usec long value.

Reply:
Here is an example. `timeout_nano_sec` (BTW, would it be more consistent to use `timeout_ns` as in Eiffel code?) is equal to `2017` nanoseconds. This corresponds to 2 microseconds. But `2017 % 1000` gives `17`. This value will be assigned to `tv_usec` instead of `2`. I believe this is wrong.